code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -1595,8 +1533,134 @@ pub mod pallet {
 		}
 	}
 }
+/// Accumulates the rewards in a pool, if the pot isn't empty. Emits the relevant events
+/// after accumulation. See [`accumulate_reward`] for more information about how the
+/// accumulation calculation is done.
+pub(crate) fn reward_accumulation_hook_reward_update_calculation<T: Config>(
+	pool_id: T::AssetId,
+	reward_asset_id: T::AssetId,
+	reward: &mut Reward<T::Balance>,
+	now_seconds: u64,
+) {
+	let pool_account = Pallet::<T>::pool_account_id(&pool_id);
+
+	log::info!("calculating rewards for pool {pool_id:?}, asset {reward_asset_id:?}");
+
+	// no need to calculate if the pot is empty, this will be unset if and when enough funds
+	// are added to the pot
+	if RewardsPotIsEmpty::<T>::contains_key(pool_id, reward_asset_id) {
+		log::info!(
+			"pot for pool {pool_id:?}, asset {reward_asset_id:?} is empty, not accumulating"
+		);
```

Comment:
These logs are not permanent, i added them to debug the accumulation logic. i would like to keep some of them however, as i feel this would be useful information to have in case something goes wrong in production (perhaps using sp-tracing)